lib/js/main.js

```diff
@@ -157,22 +160,44 @@ var sweep = (function() {
         (to.b - from.b) / steps,
         (to.a - from.a) / steps
       ];
-    } else { // space is H(U)SL
-
-      // sneaking another default in here...
-      if (space !== 'HUSL') space = args.space = 'HSL';
-
-      // convert colors to { h: _, s: _, l: _, a: _ } format
-      from = convert.toHsla(from);
-      to = convert.toHsla(to);
-
-      // deltas = [dH, dS, dL, dA]
-      deltas = [
-        0,
-        (to.s - from.s) / steps,
-        (to.l - from.l) / steps,
-        (to.a - from.a) / steps
-      ];
+    } else {
+      if (space === "HSL") {
+        // space is HSL
+        // convert colors to { h: _, s: _, l: _, a: _ } format
+        from = convert.toHsla(from);
+        to = convert.toHsla(to);
+
+        // deltas = [dH, dS, dL, dA]
+        deltas = [
+          0,
+          (to.s - from.s) / steps,
+          (to.l - from.l) / steps,
+          (to.a - from.a) / steps
+        ];
+      } else {
+        // space is HSLuv
+        from = convert.toRgba(from);
+        to = convert.toRgba(to);
+        var fromHSLuv = hsluv.rgbToHsluv([
+          from.r / 255,
+          from.g / 255,
+          from.b / 255
+        ]);
+        var toHSLuv = hsluv.rgbToHsluv([to.r / 255, to.g / 255, to.b / 255]);
+        from.h = fromHSLuv[0];
+        from.s = fromHSLuv[1] / 100;
+        from.l = fromHSLuv[2] / 100;
+        to.h = toHSLuv[0];
+        to.s = toHSLuv[1] / 100;
+        to.l = toHSLuv[2] / 100;
+        // deltas = [dH, dS, dL, dA]
```

Comment:
Great - thanks for fixing this! Since this `delta` conversion is common between HSL and HSLuv, can we move it after the `if/else` block and delete the repetition?